pkg/credentialprovider/config.go

```diff
@@ -140,19 +154,19 @@ func ReadDockerConfigJSONFile(searchPaths []string) (cfg DockerConfig, err error
 		}
 		cfg, err := readDockerConfigJsonFileFromBytes(contents)
 		if err == nil {
-			glog.V(4).Infof("found .docker/config.json at %s", absDockerConfigFileLocation)
+			glog.V(4).Infof("found %s at %s", dockerConfigJsonFileName, absDockerConfigFileLocation)
 			return cfg, nil
 		}
 	}
-	return nil, fmt.Errorf("couldn't find valid .docker/config.json after checking in %v", searchPaths)
+	return nil, fmt.Errorf("couldn't find valid %s after checking in %v", dockerConfigJsonFileName, searchPaths)
 }
-
 func ReadDockerConfigFile() (cfg DockerConfig, err error) {
-	if cfg, err := ReadDockerConfigJSONFile(nil); err == nil {
-		return cfg, nil
+	if cfg, err = ReadDockerConfigJSONFile(nil); err != nil {
+		if cfg, err = ReadDockercfgFile(nil); err != nil {
+			cfg, err = ReadDockerConfigJsonFileUsedInSecret(nil)
```

Comment:
Providing a more general method is reasonable, since actually we have had this kind of file(.dockerconfigjson) for storing docker auth in our cluster. and maybe we haven't use it directly now, a innovation of `ReadDockerConfigJsonFileUsedInSecret` make the function `ReadDockerConfigFile` more flexible, and it also will not affect the the original function.

Of course, we could also not call `ReadDockerConfigJsonFileUsedInSecret` in function `ReadDockerConfigFile`, just keep the function `ReadDockerConfigJsonFileUsedInSecret` for future reference.